File: Week_01/twoSum.py
class Solution(object):
    def twoSum(self, nums, target):
        """
        :type nums: List[int]
        :type target: int
        :rtype: List[int]
        """
        mapsum=dict()
        for i,x in enumerate(nums):
            if target-x in mapsum:
                return [mapsum[target-x],i]
            mapsum[x]=i

File: Week_01/twoSum.go
func twoSum(nums []int, target int) []int {
    mapsum:=map[int]int{}
    for i,x:=range nums{
        if k,ok := mapsum[target - x]; ok{
            return []int{k, i}
        }
        mapsum[x]=i
    }
    return nil
